crates/telio-wg/src/link_detection/mod.rs

```diff
@@ -33,47 +36,24 @@ impl LinkDetection {
         }
     }
 
-    pub fn time_since_last_rx(&self, public_key: &PublicKey) -> Option<Duration> {
+    pub fn insert(&mut self, public_key: &PublicKey, stats: Arc<Mutex<BytesAndTimestamps>>) {
         match self {
-            LinkDetection::Enabled(l) => l.time_since_last_rx(public_key),
-            LinkDetection::Disabled(l) => l.time_since_last_rx(public_key),
-        }
-    }
-
-    pub fn insert(
-        &mut self,
-        public_key: &PublicKey,
-        rx_bytes: Option<u64>,
-        tx_bytes: Option<u64>,
-        node_state: NodeState,
-    ) {
-        match self {
-            LinkDetection::Enabled(l) => l.insert(public_key, rx_bytes, tx_bytes, node_state),
-            LinkDetection::Disabled(l) => l.insert(public_key, rx_bytes),
+            LinkDetection::Enabled(l) => l.insert(public_key, stats),
```

Comment:
I believe that in case link detection is disabled -> we do not need to do anything here anymore, because `time_since_last_rx()` is not being retreived via `link_state` module anymore, but rather directly from `wg.rs`.

Which would mean, that we can delete `struct LinkDetectionDisabled` and simplify the code a bit more :)

Reply:
removed entirely